### DISCUSSION TITLE
How to output curve of strain hardening rate and strain

### DISCUSSION THREAD

**Discussion by xchengood** [idaholab/moose#19557] [Q&A Modules: Solid mechanics] [answered]
Hi Moose experts or users,

Could anyone share any ideas or thoughts about how to output the curve of strain hardening rate and strain? One example is shown below. Now I can get a reasonable curve of stress-strain. But I have no idea how to obtain the curve of strain hardening rate-strain in Moose. Thank you.
<img width="802" alt="Screen Shot 2021-12-06 at 1 19 47 PM" src="https://user-images.githubusercontent.com/74226455/144909087-593e0750-ccaa-456d-a770-6cd50e92e750.png">

**Comment by ngrilli** [comment, marked as answer]
Dear @xchengood 

Paraview has a time derivative filter, you can take the time derivative of the stress,
if you know your strain rate, you will be able to calculate the d_sigma/d_epsilon.

Best Regards,
Nicolò
